ibmcloud/cluster/vpc/main.tf

```diff
@@ -0,0 +1,80 @@
+#
+# (C) Copyright IBM Corp. 2023.
+# SPDX-License-Identifier: Apache-2.0
+#
+
+resource "ibm_is_vpc" "vpc" {
+  name = "${var.cluster_name}-vpc"
+}
+
+resource "ibm_is_floating_ip" "gateway" {
+    name = "${var.cluster_name}-gateway-ip"
+    zone = var.zone
+}
+
+resource "ibm_is_public_gateway" "gateway" {
+  name        = "${var.cluster_name}-gateway"
+  vpc         = ibm_is_vpc.vpc.id
+  zone        = var.zone
+  floating_ip = {
+    id = ibm_is_floating_ip.gateway.id
+  }
+}
+
+resource "ibm_is_subnet" "primary" {
+  name                     = "${var.cluster_name}-subnet"
+  vpc                      = ibm_is_vpc.vpc.id
+  zone                     = var.zone
+  total_ipv4_address_count = 256
+  public_gateway           = ibm_is_public_gateway.gateway.id
+}
+
+resource "ibm_is_security_group" "primary" {
+  name = "${var.cluster_name}-security-group"
+  vpc  = ibm_is_vpc.vpc.id
+}
+
+resource "ibm_is_security_group_rule" "primary_outbound" {
+  group      = ibm_is_security_group.primary.id
+  direction  = "outbound"
+  remote     = "0.0.0.0/0"
+}
+
+resource "ibm_is_security_group_rule" "primary_inbound" {
+  group      = ibm_is_security_group.primary.id
+  direction  = "inbound"
+  remote     = ibm_is_security_group.primary.id
+}
+
+resource "ibm_is_security_group_rule" "primary_ssh" {
+  group      = ibm_is_security_group.primary.id
+  direction  = "inbound"
+  remote     = "0.0.0.0/0"
+
+  tcp {
+    port_min = 22
+    port_max = 22
+  }
+}
+
+resource "ibm_is_security_group_rule" "primary_ping" {
+  group      = ibm_is_security_group.primary.id
+  direction  = "inbound"
+  remote     = "0.0.0.0/0"
+
+  icmp {
+    code = 0
+    type = 8
+  }
+}
+
+resource "ibm_is_security_group_rule" "primary_api_server" {
+  group      = ibm_is_security_group.primary.id
+  direction  = "inbound"
+  remote     = "0.0.0.0/0"
+
+  tcp {
+    port_min = 6443
```

Comment:
Based on the IKS doc here https://cloud.ibm.com/docs/containers?topic=containers-vpc-security-group&interface=ui, looks TCP ports 30000 - 32767 are used for service to service communication. Shall we also add rules for these ports? And also 80 and 443 for http/https?

Reply:
hi @huoqifeng, yes it is similar to the ibmcloud/terraform folder but simplified and for a single purpose, rather than creating a cluster, creating the podvm image, setting up the caa + demo, etc. However, we are avoiding updating the existing terraform scripts as we still want to run our nightly tests until we have the operator e2e tests in-place. Do you have a suggestion for structure/naming to future proof this without disrupting the existing e2e demo?

For the purpose of testing the operator I don't think any additional services ports need to be exposed as it mostly internal communication, but maybe 80 and 443 would be useful for some test scenarios. When we moved to IKS clusters we will be able to setup ingress traffic.

Reply:
OK, if the self-managed cluster is not to be used for service to service communication test. I'm fine.